packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/RealmSyncUtils.kt

```diff
@@ -24,18 +25,41 @@ import io.realm.kotlin.mongodb.exceptions.UserAlreadyExistsException
 import io.realm.kotlin.mongodb.exceptions.UserNotFoundException
 import io.realm.kotlin.mongodb.exceptions.WrongSyncTypeException
 import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ChannelResult
 
 internal fun <T, R> channelResultCallback(
     channel: Channel<Result<R>>,
     success: (T) -> R
 ): AppCallback<T> {
     return object : AppCallback<T> {
         override fun onSuccess(result: T) {
-            channel.trySend(Result.success(success.invoke(result)))
+            try {
+                val sendResult: ChannelResult<Unit> = channel.trySend(Result.success(success.invoke(result)))
```

Comment:
I ran into a bug during development where the callback would just hang. This happened because I messed up the return types from the C-API and the exception was just swallowed.

This change makes sure that errors during callback are propagated back to the user.